screens/Onboarding.tsx

```diff
@@ -0,0 +1,105 @@
+import Slide from '@/components/Onboarding/Slide';
+import PageIndicator from '@/components/PageIndicator';
+import Button from '@/components/ui/Button';
+import { ONBOARDING_DATA, OnboardingStep } from '@/constants/onboardingData';
+import { useOnboardingStore } from '@/stores/onboarding/store';
+import { saveOnboardingStatus } from '@/stores/onboarding/utils/storage';
+import { replaceToHome } from '@/utilities/navigationHelpers';
+import React, { useRef } from 'react';
+import {
+	Dimensions,
+	FlatList,
+	NativeScrollEvent,
+	NativeSyntheticEvent,
+	StyleSheet,
+	View,
+} from 'react-native';
+import { SafeAreaView } from 'react-native-safe-area-context';
+
+const { width: SCREEN_WIDTH } = Dimensions.get('window');
+const PADDING = 24;
+const SLIDE_WIDTH = SCREEN_WIDTH - PADDING * 2;
+
+const Onboarding = () => {
+	const currentStep = useOnboardingStore((state) => state.currentStep);
+	const setCurrentStep = useOnboardingStore((state) => state.setCurrentStep);
+	const setHasCompletedOnboarding = useOnboardingStore((state) => state.setHasCompletedOnboarding);
+
+	const flatListRef = useRef<FlatList>(null);
+	const currentData = ONBOARDING_DATA[currentStep] || ONBOARDING_DATA[0];
+
+	const handleNext = async () => {
+		const lastStep = ONBOARDING_DATA.length - 1;
+
+		if (currentStep < lastStep) {
+			const nextStep = currentStep + 1;
+
+			setCurrentStep(nextStep);
+			flatListRef.current?.scrollToIndex({ index: nextStep, animated: true });
+		} else {
+			await handleCompleteOnboarding();
+		}
+	};
+
+	const handleCompleteOnboarding = async () => {
+		await saveOnboardingStatus();
+		setHasCompletedOnboarding(true);
+
+		replaceToHome();
+	};
+
+	const handleScroll = (event: NativeSyntheticEvent<NativeScrollEvent>) => {
+		const offsetX = event.nativeEvent.contentOffset.x;
+		const index = Math.round(offsetX / SLIDE_WIDTH);
+
+		setCurrentStep(index);
+	};
+
+	const renderSlide = ({ item, index }: { item: OnboardingStep; index: number }) => {
+		const isLast = index === ONBOARDING_DATA.length - 1;
+
+		return <Slide item={item} width={SLIDE_WIDTH} showBadge={isLast ? false : true} />;
+	};
+
+	return (
+		<SafeAreaView style={styles.screen} edges={['bottom']}>
+			<View style={[styles.container, { padding: PADDING }]}>
+				{/* 슬라이드 가능한 콘텐츠 영역 */}
+				<FlatList
+					ref={flatListRef}
+					data={ONBOARDING_DATA}
+					keyExtractor={(item) => item.step.toString()}
+					renderItem={renderSlide}
+					horizontal
+					pagingEnabled
+					showsHorizontalScrollIndicator={false}
+					onScroll={handleScroll}
```

Comment:
```suggestion
					onMomentumScrollEnd={handleScroll}
```

클로드코드가 추천해준거라 저도 정확히 알고쓰는 것은 아니지만.. 제 슬라이드에는 `onMomentumScrollEnd`을 사용해서 index를 잡도록 되어있네요. 아마도 관성으로 스크롤이 멈추는 시점과 지점을 기준으로 잡는 것 같은데, `<FlatList />` 기본 동작도 paging에 관성을 반영하고 있을테니 이렇게 해주는게 좋을 것 같습니다.

관성으로 세게 튕겨보면(?) 문제가 있는지 없는지 알 수 있을것같아요.

Reply:
간단하게 이벤트 호출횟수를 각각 콘솔에 출력해보니 `onScoll`은 스크롤하는 동안 프레임마다 호출되고 `onMomentumScollEnd`은 스크롤이 종료된 시점에 1번만 호출되더라구요.
찾아보니 `onScroll`은 애니메이션처럼 실시간으로 UI 업데이트가 필요한 경우에 적합하고, 지금처럼 스크롤이 끝난 뒤에만 처리하면 되는 상황에서는 말씀하신대로`onMomentumScollEnd`를 사용하는 게 맞는 것 같습니다.
덕분에 또 하나 배우게 되었습니다! 🙇

Reply:
아 그렇군요, 이유 공유해주셔서 감사합니다, 저도 배웠습니다. ㅎㅎ 🙇‍♂️